src/cephadm/cephadm.py

```diff
@@ -9746,6 +9799,23 @@ def _get_parser():
         '--allow-overwrite',
         action='store_true',
         help='allow overwrite of existing --output-* config/keyring/ssh files')
+    # following logic to have both '--keep-cluster-files' and '--no-keep-cluster-files'
+    # has been included in argparse of python v3.9, however since we have to support
+    # older python versions the following is more generic. Once python v3.9 becomes
+    # the minium supported version we can implement the same by using the new option
+    # argparse.BooleanOptionalAction
+    group = parser_bootstrap.add_mutually_exclusive_group()
+    group.add_argument(
+        '--keep-cluster-files',
+        action='store_true',
+        default=False,
+        help='Do not delete cluster in case of a failed installation')
+    group.add_argument(
+        '--no-keep-cluster-files',
```

Comment:
NIT, these flag names are probably adequate, but what do you think about naming them `--cleanup-on-failure` and `--no-cleanup-on-failure`? I feel like those names would be a bit more clear on what these flags are actually doing.

Reply:
I thinks the proposed names for the flags are more adequate :+1: